common/step_provision.go

```diff
@@ -46,6 +46,17 @@ func PopulateProvisionHookData(state multistep.StateBag) map[string]interface{}
 		// Warn user that the id isn't implemented
 		hookData["ID"] = "ERR_ID_NOT_IMPLEMENTED_BY_BUILDER"
 	}
+
+	// source_ami_name is placed in state by the amazon builder
+	// and it's not implemented by other builders
+	amiName, ok := state.GetOk("source_ami_name")
+	if ok {
+		hookData["SourceAMIName"] = amiName
```

Comment:
I think `SourceAMIName` is super specific to the amazon builders, I wonder if we can pick something less specific that other builders could use.  This would be used in templates like so :

```
{{ builder 'SourceAMIName' }}
```

 The word 'ami' is super specific to amazon and may we could store this value under "SourceImageName" like so :

```suggestion
		hookData["SourceImageName"] = sourceImageName
```
So that this is already defined for other builders 🤔.

Reply:
Nice!! I will do it this way!